app/controllers/admin/users_controller.rb

```diff
@@ -13,24 +13,30 @@ def update
     private
 
     def user_params
-      accessible = %i[
+      accessible = accessible_params
+      accessible << %i[password password_confirmation] if params[:user][:password].present?
+      verify_usernames params.require(:user).permit(accessible)
+    end
+
+    # make sure usernames are not empty, to be able to use the database unique index
+    def verify_usernames(user_params)
+      user_params[:twitter_username] = nil if user_params[:twitter_username] == ""
+      user_params[:github_username] = nil if user_params[:github_username] == ""
+      user_params
+    end
+
+    def accessible_params
+      attributes_params | url_params | other_params
```

Comment:
I'm not very keen on these names: all params are attributes, maybe `core_params` is a better name?

Reply:
I agree with you